### PR TITLE
Add missing injections

### PR DESCRIPTION
**Pull request by ndr-brt**
Added missing injections on `FederatedCatalogCacheExtension` that caused NPE on extension initialization.
Removed a duplicated `publishing` block on the extension build file (rebase typo?)

### CODE REVIEW

File: extensions/azure/fcc-node-directory-cosmos/build.gradle.kts [hunk 1]
```diff
@@ -36,12 +36,4 @@ publishing {
...
-}
-publishing {
-    publications {
-        create<MavenPublication>("in-memory.asset-resolver") {
-            artifactId = "in-memory.asset-resolver"
-            from(components["java"])
-        }
-    }
-}
+}
```
Comment (paullatzelsperger) [inline extensions/azure/fcc-node-directory-cosmos/build.gradle.kts:39]: can you change this to `fcc-node-directory-cosmos`? we'll need that maven artifact.

**Review by paullatzelsperger — CHANGES_REQUESTED** [review]

File: extensions/azure/fcc-node-directory-cosmos/build.gradle.kts (hunk 1 above)
Comment (ndr-brt) [reply extensions/azure/fcc-node-directory-cosmos/build.gradle.kts:39]: is already like that: https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/0bef2f8aecb01c9a2d9b33d46b7bbb084fc5dad8/extensions/azure/fcc-node-directory-cosmos/build.gradle.kts#L34

Comment (paullatzelsperger) [reply extensions/azure/fcc-node-directory-cosmos/build.gradle.kts:39]: ok, sorry, i must have gotten an out dated version. will approve then.

**Review by paullatzelsperger — APPROVED** [review]

**Comment by jimmarino** [comment]
Closing as these changes were included in #486.

**Pull request closed without merging**